### PR TITLE
statsd: don't fetch resource on each flush()

### LINKED ISSUE #375: statsd: don't get resource everytime we flush
we shouldn't need to get resource and it's metrics every time we flush. just save it.

https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/statsd.py#L91-L93

### PR DESCRIPTION
**Pull request by jd**
Fixes #375

### CODE REVIEW

**Comment by hguemar** [comment]
Change do fix the issue. 👍

**Comment by jd** [comment]
Pull-request updated, HEAD is now e80867ed8ebc49e68e74c141ec788ef907eab3a7

File: gnocchi/statsd.py [hunk 1]
```diff
@@ -110,6 +113,7 @@ def flush(self):
                         archive_policy_name=ap_name,
                         name=metric_name,
                         resource_id=self.conf.statsd.resource_id)
```
Comment (chungg) [inline gnocchi/statsd.py]: can do this in another patch but we should try to catch NamedMetricAlreadyExists if someone creates metric via api

**Comment by jd** [comment]
Pull-request updated, HEAD is now 9756cb2f0e5af79d87899132c7e7d4dd9cd20f8e
